lib/apiservers/engine/backends/volume.go

```diff
@@ -211,15 +221,44 @@ func validateDriverArgs(args map[string]string, model *models.VolumeRequest) err
 		return nil
 	}
 
+	//check if it is just a numerical value
+	validator, _ := regexp.Compile("^[0-9][0-9]*$") //note this is a global line match. Find will return 1 or no matches.
+	match := validator.Find([]byte(capstr))
+	if match != nil {
+		//input has no units in this case.
+		capacity, err := strconv.ParseInt(capstr, 10, 64)
+		if err != nil {
+			return fmt.Errorf("Invalid size: %s", capstr)
+		}
+		model.Capacity = capacity
+		return nil
+	}
+
 	capacity, err := units.FromHumanSize(capstr)
 	if err != nil {
 		return err
 	}
+
+	if capacity < 1 {
+		return fmt.Errorf("Capacity value too large: %s", capstr)
```

Comment:
It indicates and overflow from `FromHumanSize` they do not return an error with it as far as I can tell... it actually will return a negative number. I may need to do an additional error check.